src/components/utils/caret.ts

```diff
@@ -112,7 +112,7 @@ export function checkContenteditableSliceForEmptiness(contenteditable: HTMLEleme
  * @param input - input where caret should be checked
  */
 export function isCaretAtStartOfInput(input: HTMLElement): boolean {
-  const firstNode = $.getDeepestNode(input);
+  const firstNode = $.isNativeInput(input) ? input : $.getDeepestNode(input);
```

Comment:
I'd suggest to update `getDeepestNode` instead. It already contains a logic that will return a passed node, but seems like `node[child]` statement does not cover this case

Reply:
Thank you for pointing it out.
I think you are right too.

Fixed in [8cec5c9](https://github.com/codex-team/editor.js/pull/2836/commits/8cec5c9a760d8d033df0196e2aff242d97ae6c70).

Reply:
@neSpecc 
Please re-review.